packages/dds/map/src/directory.ts

```diff
@@ -661,16 +662,13 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
      * {@inheritDoc @fluidframework/shared-object-base#SharedObject.loadCore}
      */
     protected async loadCore(storage: IChannelStorageService) {
-        const blob = await storage.readBlob(snapshotFileName);
-        const header = bufferToString(blob, "utf8");
-        const data = JSON.parse(header);
+        const data = await readAndParse(storage, snapshotFileName);
         const newFormat = data as IDirectoryNewStorageFormat;
         if (Array.isArray(newFormat.blobs)) {
             // New storage format
             this.populate(newFormat.content);
             await Promise.all(newFormat.blobs.map(async (value) => {
-                const newBlob = await storage.readBlob(value);
-                const dataExtra = JSON.parse(bufferToString(newBlob, "utf8"));
+                const dataExtra = await readAndParse(storage, value);
```

Comment:
Same here. It's not ideal, but if it's common, you might consider changing API to
- accept (optional) default value (for case where blob is not there)
- throwing if blob is not there and optional value was not provided.

I.e. instead of 
  const content = await readAndParse<ICounterSnapshotFormat>(storage, snapshotFileName) ?? { value: 0 };
Have

  const content = await readAndParse<ICounterSnapshotFormat>(storage, snapshotFileName, { value: 0 });

Again, not sure if it's good idea, just sharing some possibilities.

Reply:
I believe readAndParse() (as well as old code) can never return undefined. So I think this code is Ok as is, but all the other code that has IFs (before and after your change) is wrong in creating a feeling that undefined is possible return result - it's not! So I'd rather remove all IFs (see comment above)